src/erlcloud_s3.erl

```diff
@@ -1227,6 +1232,228 @@ set_bucket_attribute(BucketName, AttributeName, Value, Config)
+        {schedule, "Schedule", fun extract_schedule/1},
+        {included_object_versions, "IncludedObjectVersions", text},
+        {optional_fields, "OptionalFields", fun extract_optional_fields/1}
+    ],
+    erlcloud_xml:decode(Attributes, Content).
+
+extract_destination([Destination]) ->
+    Attributes = [{s3_bucket_destination, "S3BucketDestination", fun extract_s3_bucket_destination/1}],
+    erlcloud_xml:decode(Attributes, Destination).
+
+extract_s3_bucket_destination([S3BucketDestination]) ->
+    Attributes = [
+        {format, "Format", text},
+        {account_id, "AccountId", text},
+        {bucket, "Bucket", text},
+        {prefix, "Prefix", text}
+    ],
+    erlcloud_xml:decode(Attributes, S3BucketDestination).
+
+extract_schedule([Schedule]) ->
+    Attributes = [{frequency, "Frequency", text}],
+    erlcloud_xml:decode(Attributes, Schedule).
+
+extract_filter([Filter]) ->
+    Attributes = [{prefix, "Prefix", text}],
+    erlcloud_xml:decode(Attributes, Filter).
+
+extract_optional_fields([OptionalFields]) ->
+    Attributes = [{field, "Field", list}],
+    erlcloud_xml:decode(Attributes, OptionalFields).
+
+-spec get_bucket_inventory(string(), string()) ->
+    {ok, Result:: list(term())} | {error, Reason::term()}.
+get_bucket_inventory(BucketName, InventoryId) when is_list(BucketName), is_list(InventoryId) ->
+    get_bucket_inventory(BucketName, InventoryId, default_config()).
+
+-spec get_bucket_inventory(string(), string(), aws_config()) ->
+    {ok, Result:: list(term())} | {error, Reason::term()}.
+get_bucket_inventory(BucketName, InventoryId, #aws_config{} = Config)
+    when is_list(BucketName), is_list(InventoryId) ->
+
+    Params = [{"id", InventoryId}],
+    case s3_request2(Config, get, BucketName, "/", "inventory", Params, <<>>, []) of
+        {ok, {_Headers, Body}} ->
+            {ok, extract_inventory_configuration(element(1, xmerl_scan:string(binary_to_list(Body))))};
+        Error ->
+            Error
+    end.
+
+-spec put_bucket_inventory(string(), list()) -> ok | {error, Reason::term()}.
+put_bucket_inventory(BucketName, Inventory)
+    when is_list(BucketName), is_list(Inventory) ->
+
+    put_bucket_inventory(BucketName, default_config()).
+
+-spec put_bucket_inventory(string(), list(), aws_config())
+        -> ok | {error, Reason::term()}.
+put_bucket_inventory(BucketName, Inventory, #aws_config{} = Config)
+    when is_list(BucketName), is_list(Inventory) ->
+
+    InventoryId = proplists:get_value(id, Inventory),
+    XmlInventory = encode_inventory(Inventory),
+    put_bucket_inventory(BucketName, InventoryId, list_to_binary(XmlInventory), Config).
+
+-spec put_bucket_inventory(string(), string(), binary(), aws_config())
+        -> ok | {error, Reason::term()}.
+put_bucket_inventory(BucketName, InventoryId, XmlInventory, #aws_config{} = Config)
+    when is_list(BucketName), is_list(InventoryId), is_binary(XmlInventory) ->
+    Md5 = base64:encode(crypto:hash(md5, XmlInventory)),
```

Comment:
is it really needed i do not see it in docs. http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketPUTInventoryConfig.html

isn't SigV4 doing it all for you?